### PR TITLE
feat: Fetch epoch info from the RewardDistribution contract

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/lib/epoch-info.ts [hunk 1]
```diff
@@ -31,51 +35,114 @@ export const StakingSettings: {
...
+    }
+  }
+
+  parseEpochDetailsLog (event: any): [number, EpochDetailsInfo] {
+    const { blockNumber, decodedLog } = event
+    return [
+      decodedLog.args.epoch.toString(),
+      {
+        endBlockNumber: blockNumber,
+        calcTimeStamp: 1,
```
Comment (ColonelJ) [inline src/lib/epoch-info.ts:129]: The calcTimeStamp is stored on chain and needs to be retrieved for each epoch.

**Review by ColonelJ — CHANGES_REQUESTED** [review]
Part of the information for each epoch is stored on chain, including the number of epochs and calc timestamps and send blocknumber/timestamps for each epoch.  Some of this info is required (calc timestamp), and anyway it might be better for the full node if you search for the event only on the individual blocks where they occur rather than a large range of blocks.  Somehow there needs to be a check that you got the events for all the epochs you expect.  Also the `CurrentEpoch` in `StakingSettings` can probably be set based on the number of epochs read from the SC.

**Comment by shresthagrawal** [comment]
I agree with Jason's comments! You can follow the following sequence: 
1. Fetch the currentEpoch
2. For each epoch index before the currentEpoch
    1. Fetch the epochHistory[i] to get sendBlockNumber, and calcTimestamp
    2. Use sendBlockNumber to fetch for exact event 
    3. Set the all the info about the epoch
    - If anything in this level fails log an error, retry n times and finally throw an error
- if you get any error simply catch, log and process.exit

**Review by ColonelJ — CHANGES_REQUESTED** [review]
You still need to update `CurrentEpoch` in `StakingSettings`.

**Review by ColonelJ — APPROVED** [review]

**Pull request merged**